src/tests/PerfCompareHeader.test.tsx

```diff
@@ -0,0 +1,44 @@
+import userEvent from '@testing-library/user-event';
+
+import PerfCompareHeader from '../components/Shared/PerfCompareHeader';
+import { setAlert } from '../reducers/AlertSlice';
+import type { AlertType } from '../types/state';
+import { render, screen, store } from './utils/test-utils';
+
+const message = 'Maximum 4 Revisions';
+const severity: AlertType['severity'] = 'error';
+let title;
+
+describe('PerfCompareHeader', () => {
+  it('renders correctly when there is no alert', () => {
+    render(<PerfCompareHeader />);
+
+    expect(screen).toMatchSnapshot();
+  });
+
+  it('should hide alert when close button is clicked', async () => {
+    // set delay to null to prevent test time-out due to useFakeTimers
+    const user = userEvent.setup({ delay: null });
+    title = undefined;
+
+    store.dispatch(setAlert({ message, severity, title }));
+    render(<PerfCompareHeader />);
+
+    expect(screen.getByText('Maximum 4 Revisions')).toBeInTheDocument();
+    expect(screen).toMatchSnapshot();
```

Comment:
Usually I use `document.body` to match.
Looking at the new snapshot, I think that using `screen` doesn't work as you expect :-)